### PR TITLE
Development: migrate frontend tests to playwright

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

[Playwright](https://playwright.dev) seems to be the new leader in frontend e2e testing and Im pretty convinced. It's easier to write tests and the network output can be recorded (HAR files). Also integration with VS code is great. At the moment this only tests against chromium (like we're currently doing) but you can also run it against many browsers quite easily.

I've replicated all the existing tests and added some more. Certainly the hope is to keep adding eg as frontend features are implemented, but clearly frontend testing just isn't as robust as backend.

The only downside I see is that cypress gave very nice output during Actions runs where as this doesn't, it just sorta hangs and then spits out the results. I would welcome any thoughts on the ci part.

Screenshot 2023-05-13 at 11 06 26 PM

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by shamoon** [comment]
Definitely one area for improvement (that I don’t exactly understand the cause of): sometimes the ci fails to actually start the frontend for testing https://github.com/paperless-ngx/paperless-ngx/actions/runs/4982355359/jobs/8917910726

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -192,10 +192,20 @@ jobs:
           node-version: ${{ matrix.node-version }}
           cache: 'npm'
           cache-dependency-path: 'src-ui/package-lock.json'
-      - run: cd src-ui && npm ci
-      - run: cd src-ui && npm run lint
-      - run: cd src-ui && npm run test
-      - run: cd src-ui && npm run e2e:ci
+      - name: Install dependencies
```
Comment (stumpylog) [inline .github/workflows/ci.yml:195]: Now that these include a name, maybe include the blank line for a little visual differentiation?

Comment (shamoon) [reply .github/workflows/ci.yml]: Sure!

**Comment by shamoon** [comment]
Here's issue I mentioned, now that Im looking at it perhaps has to do with two runs at the same time?
https://github.com/paperless-ngx/paperless-ngx/actions/runs/5016598231/jobs/8993672873
vs
https://github.com/paperless-ngx/paperless-ngx/actions/runs/5016597679/jobs/8993669466

**Comment by shamoon** [comment]
Ok hopefully smoothed this out, tweaked some settings from https://github.com/pdyck/short-link-management/blob/main/playwright.config.ts according to https://github.com/microsoft/playwright/issues/16906

🤞

**Comment by stumpylog** [comment]
I see there's no reports to upload: https://github.com/paperless-ngx/paperless-ngx/actions/runs/5016785508/jobs/8994092965#step:8:10

It probably doesn't matter much, but maybe the folder isn't the right one?

**Comment by shamoon** [comment]
> It probably doesn't matter much, but maybe the folder isn't the right one?

Ah, youre right of course, great catch thanks. Fixed now: https://github.com/paperless-ngx/paperless-ngx/actions/runs/5018133579/jobs/8997165439

**Review by stumpylog — APPROVED** [review]
CI stuff looks good.  I'll defer to you on the testing parts

**Comment by shamoon** [comment]
Cool. Yea my hope is this can help to continue improving frontend testing moving forward.

**Pull request merged**